hibernate-core/src/main/java/org/hibernate/dialect/SpannerDialect.java

```diff
@@ -114,6 +119,15 @@ public SpannerDialect(DialectResolutionInfo info) {
 		registerKeywords( info );
 	}
 
+	@Override
+	public int getMaxVarcharLength() {
+		return 2_621_440;
```

Comment:
Wow, I relearned something.  Totally forgot you can format numbers this way! :laughing: